### ISSUE TITLE
deviceName not updating when user updates device name on device

### ISSUE THREAD

**Issue by chrisdrackett** [react-native-device-info/react-native-device-info#1272]
## Summary

|             |     |
| ----------- | --- |
| Version     | 8.1.4   |
| Affected OS | iOS   |
| OS Version  | 14.7   |

## Current behavior

after changing the device name and then calling `DeviceInfo.getDeviceName()` I'm still getting the old value.

1. open app (device name reports as `iPhone`)
2. close app and change device name in settings to `iPhone new`
3. open app and call `DeviceInfo.getDeviceName()`
4. returns `iPhone` and not `iPhone new`

## Expected behavior

`DeviceInfo.getDeviceName()` should report the current device name

**Comment by mikehardy** [comment]
That's odd. Looks like we fetch it dynamically:

https://github.com/react-native-device-info/react-native-device-info/blob/0e1319c69532b2bed3f5e8ba47f778abc89395ee/ios/RNDeviceInfo/RNDeviceInfo.m#L146-L157

How did it go when you reached in there and added some debugging print statements + watched the logs? Is the system API we use behaving correctly? Does it mention anything in the related documentation about caching behavior in the system?

**Comment by schie** [comment]
@mikehardy in the js/ts side, it is memo-ized. We can remove the memo to resolve this issue.

**Comment by mikehardy** [comment]
Ah ha! @chrisdrackett if you post a PR removing the memoization there, can happily merge + release

tangent: oh man, the semantic release bot is golden here @schie, thank you so much for that

**Comment by anujwits** [comment]
This issue is still persist for android  devices.
Android Version: 11
react-native-device-info: 8.7.0

Please suggest a workaround.

**Comment by anujwits** [comment]
Behaviour for this  is strange, this is working on some devices and not for others.
Do we need to give some permission at device level?

**Comment by mikehardy** [comment]
previously needed bluetooth, not supposed to need it now

**Comment by anujwits** [comment]
Yes no bluetooth permission is required. This work's for the first time.
If i update phone name, then updated name is not reflecting.

**Comment by mikehardy** [comment]
@anujwits then this comment drives progress on your part:

> How did it go when you reached in there and added some debugging print statements + watched the logs? Is the system API we use behaving correctly? Does it mention anything in the related documentation about caching behavior in the system?

**Comment by jessenovotny** [comment]
Does not work on iOS with v10.4.0. No matter what name our devices getDeviceName() always returns `iPad`. I'm not much of a iOS developer so unsure the best way to troubleshoot. Please advise

**Comment by jessenovotny** [comment]
Turns out this is an issue with iOS 16. For anyone who winds up here, please refer to the issue documented here: https://github.com/react-native-device-info/react-native-device-info/issues/1516
